third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -122,6 +122,66 @@ Value shflIdxSync(Location loc, ConversionPatternRewriter &rewriter, Value val,
                         i32_val(0x1f));
 }
 
+Value storeShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+                  Value val, Value pred) {
+  auto ty = val.getType();
+  auto val_ty = vec_ty(ty, 1);
+  Value vec_val = undef(val_ty);
+  vec_val = insert_element(val_ty, vec_val, val, i32_val(0));
```

Comment:
You can combine two lines here:

Value vec_val = insert_element(val_ty, undef(val_ty), val, i32_val(0)))

I think this is easier to read because there's fewer variables and those variables aren't changing.  (Constancy makes code easier.)  Same below.